platform/core/src/studies/services/wado/retrieveMetadata.js

```diff
@@ -468,37 +468,70 @@ async function createStudyFromSOPInstanceList(server, sopInstanceList) {
  * @param {string} studyInstanceUid The Study Instance UID of the study which needs to be loaded
  * @returns {Object} A study descriptor object
  */
-async function RetrieveMetadata(server, studyInstanceUid) {
-  return (server.enableStudyLazyLoad !== false
+async function RetrieveMetadata(server, studyInstanceUid, filters) {
+  return (server.enableStudyLazyLoad !== false && false
     ? lazyLoadStudyMetadata
-    : loadStudyMetadata)(server, studyInstanceUid);
+    : loadStudyMetadata)(server, studyInstanceUid, filters);
 }
 
 /**
  *
  * @param {*} server
  * @param {*} studyInstanceUID
  */
-async function loadStudyMetadata(server, studyInstanceUID) {
-  const dicomWeb = new api.DICOMwebClient({
+async function loadStudyMetadata(server, studyInstanceUID, filters) {
+  const client = new api.DICOMwebClient({
     url: server.wadoRoot,
     headers: DICOMWeb.getAuthorizationHeader(server),
   });
-  return dicomWeb
-    .retrieveStudyMetadata({ studyInstanceUID })
-    .then(result => createStudyFromSOPInstanceList(server, result));
+
+  const options = {
+    studyInstanceUID,
+  };
+
+  const { seriesInstanceUID } = filters;
+
+  let retrieveMethod = client.retrieveStudyMetadata;
+
+  if (seriesInstanceUID) {
+    retrieveMethod = client.retrieveSeriesMetadata;
+    options['seriesInstanceUID'] = seriesInstanceUID;
+  }
+
+  let studyMetadata;
+
+  try {
+    studyMetadata = await retrieveMethod.call(client, options);
+  } catch (e) {
+    // in case occurred error when retrieveSeriesMetadata, retrieve again from retrieveStudyMetadata
+    if (seriesInstanceUID) {
+      studyMetadata = await client.retrieveStudyMetadata({ studyInstanceUID });
+    } else {
+      throw e;
+    }
+  }
```

Comment:
Wondering if we should clean this up a little to make it clearer what's going on?

0. Identify request + options
1. Primary request
2. Fallback request
3. ???
4. Profit

Reply:
ok changed a bit the approach